src/Control/Effect/Accum.hs

```diff
@@ -0,0 +1,63 @@
+{- | An effect allowing writes to an accumulated quantity alongside a computed value,
+and reads from the accumulator.
+An 'Accum' @w@ effect keeps track of a monoidal datum of type @w@ and strictly appends to that monoidal value with the 'add' effect.
+Previous writes to that value can be read with the 'look' effect.
+
+Predefined carriers:
+
+* "Control.Carrier.Accum.Church"
+* "Control.Carrier.Accum.Strict". (A lazy carrier is not provided due to the inherent space leaks associated with lazy accumulation monads, similar to lazy writer monads.)
+* "Control.Monad.Trans.Accum"
+* If 'Accum' @w@ is the last effect in a stack, it can be interpreted to a function @w -> (w, a)@ given some result type @a@ and the presence of a 'Monoid' instance for @w@.
+
+-- | @since 1.0.3.0 -- FIXME Is this right?
+-}
+
+-- FIXME Why not haddockish single quotes?
+
+module Control.Effect.Accum
+( -- * Accumulation effect
+  Accum(..)
+, add
+, look
+, looks
+  -- * Re-exports
+, Algebra
+, Has
+, run
+) where
+
+import Control.Algebra
+-- FIXME Why have an internal module if we reexport everything?
```

Comment:
This is necessary to avoid cyclic imports given the instance defined in `Control.Algebra`. Notably, we only do it for effects when we define carriers for types defined elsewhere—e.g. `transformers` or `base`.

We used hs-boot files to break the cycles in the past, but found them to be difficult to maintain, & `cabal`/`ghc`’s support for them was buggy.